c++/src/kj/string.h

```diff
@@ -554,6 +554,107 @@ template <typename T, typename = decltype(toCharSequence(instance<T&>()))>
 inline auto tryToCharSequence(T* value) { return kj::toCharSequence(*value); }
 inline StringPtr tryToCharSequence(...) { return "(can't stringify)"_kj; }
 // SFINAE to stringify a value if and only if it can be stringified.
+
+template <typename T>
+class HasCappedCapacity_ {
+  using yes = char;
+  using no = yes[2];
+
+  template<typename C> static auto test(void*) -> decltype(C::capacity(), yes{});
+  template<typename C> static no& test(...);
+
+public:
+  static constexpr bool value = sizeof(test<T>(nullptr)) == sizeof(yes);
+};
+template <typename T>
+class HasCappedCapacity_<SignalSafeCharSequence<T>>: public HasCappedCapacity_<T> {};
+template <typename T, size_t N>
+class HasCappedCapacity_<T(&)[N]> { static constexpr bool value = true; };
+template <typename T, size_t N>
+class HasCappedCapacity_<const T(&)[N]> { static constexpr bool value = true; };
```

Comment:
There's a bug here in that T should only be char since the capacity required to stringify int[N] is unknown (depends on the delimiter used & there's no special variant of `Delimiter` that has a known capacity for when we use the default `, ` delimiter).

When I remove that, I need both forms (added static asserts).

Reply:
Err... well HasCappedCapacity_<T[N]> can just disappear altogether because it's checking after `toCharSequence`.